[READY]Adds new Space Ruin with ghost roles.

EgorDinamit opened the pull request:
## About the pull request

- Adds new syndicate space ruin with 5 ghost role spawners. Can be described as smaller and _cooler_ Charlie Station. 
Contains indestructible vault with a blast-door that opens when special randomly generated code is spoken near it. Paper with said code spawns in the captain's backpack.
_First, **unprotected** part_ of the vault contains basic engineering tools, solar panels crate, healing items, mining tools, ORM, Ore Silo and some donk co. toys.
_Second, **protected part, with indestructible walls and voice-activated blast-door**,_ contains two Cybersun Hardsuits, One Energy sword, One deluxe code-speak manual and refined materials(50 metal, 50 glass, 50 plastitanium).


- Probably fixed Nanotrasen NPCs (they were leaving invisible simplemob corpse on death), removed police-speak from the main type and added few more subtypes, including assault and elite officers for the ruin.
- Added "Cybersun industries" hardsuit. Gives high(~70) protection against laser-based weapons and small(~25) protection against all the other types of damage.

![Screenshot_40](https://user-images.githubusercontent.com/53223414/74470007-1d8d4280-4eb7-11ea-98b5-e2ca2f1ea9be.png)
![Screenshot_41](https://user-images.githubusercontent.com/53223414/74469997-19f9bb80-4eb7-11ea-9cac-89ff08c8c882.png)
![Screenshot_42](https://user-images.githubusercontent.com/53223414/74520460-0ab75400-4f31-11ea-9ce9-9764c4d6c9d9.png)


## Why It's Good For The Game

Sometimes space feels empty and lacks variations of ruins, I think the same goes for ghost roles.
The ship can't be moved and doesn't contain that much of overpowered loot, so it would be hard to use it as grief tool. The point of said ghost roles is to fix the power, reinforce the ship and secure the syndicate secret docs.

It's always good to give players more various roles to play, especially that goes for players that were permanently removed from the round, giving them an opportunity to start as a new being, instead of sitting as a ghost for the next hour.

## Changelog
:cl: EgorDinamit
add: Long-range scanners found unknown armed ship near the station. Caution should be taken while investigating the object.
add: Recent operations discovered syndicate storage pods all around the universe. It seems that in your location might be one, containing experimental hardsuit made by Cyber Sun industries long ago.
add: Discovering that there is no need to use soldiers as security officers, Nanotrasen hired most professional units into the new formed assault force.
imageadd: Added icons for Cybersun hardsuit.
imageadd: Added icons for new NT NPCs subtypes: Assault and Elite(ERT suit) officers.
config: Added forgottenship ruin to spaceruinblacklist config.
/:cl:

EgorDinamit committed:
Add files via upload

EgorDinamit committed:
Update spaceruinblacklist.txt

EgorDinamit committed:
Update space.dm

EgorDinamit committed:
Updating forgotten ship ruin

EgorDinamit committed:
Updated all needed files

EgorDinamit committed:
Merge pull request #1 from tgstation/master

Updating fork, because I don't know other ways.

EgorDinamit committed:
Update-update

EgorDinamit committed:
Merge pull request #2 from tgstation/master

Updating my own repo.

EgorDinamit committed:
Update

EgorDinamit committed:
Probably the last update

EgorDinamit committed:
Updated even further!

EgorDinamit committed:
ruin code update

EgorDinamit committed:
Final update, ha-ha

EgorDinamit committed:
oh god, more updates

EgorDinamit committed:
E

EgorDinamit committed:
Delete suit.dmi

EgorDinamit committed:
Delete head.dmi

EgorDinamit committed:
Delete hats.dmi

EgorDinamit committed:
Delete suits.dmi

EgorDinamit committed:
Merge branch 'master' of git://github.com/tgstation/tgstation into tgstation-master

EgorDinamit committed:
Merge branch 'tgstation-master'

EgorDinamit committed:
E

EgorDinamit committed:
E

EgorDinamit committed:
Update forgottenship.dm

EgorDinamit committed:
Updated map once again

EgorDinamit committed:
Update forgottenship.dmm

EgorDinamit committed:
Update forgottenship.dmm

EgorDinamit committed:
Update tgstation.dme

EgorDinamit committed:
Update ghost_role_spawners.dm

YakumoChen commented:
>door

>impossible to get through

...is the room running on magic power at least or do you know you can just crowbar depowered blast doors?

EgorDinamit commented:
> > door
> 
> > impossible to get through
> 
> ...is the room running on magic power at least or do you know you can just crowbar depowered blast doors?

Protected vault is running on the infinite power, yep.

YakumoChen commented:
does the whole room (including the walls) also block teleporting and other common means to ignore the concept of walls?

EgorDinamit committed:
Fixed most errors

EgorDinamit commented:
> does the whole room (including the walls) also block teleporting and other common means to ignore the concept of walls?

I'm kinda new to it and I don't know how to prevent people from teleporting in here.

EgorDinamit committed:
Map Merger used.

EgorDinamit committed:
Adds noteleport flag to vault.

EgorDinamit commented:
> does the whole room (including the walls) also block teleporting and other common means to ignore the concept of walls?

Okay, added "noteleport" to vault, now it should be fine, I guess.

Ratvar commented:
So you just spawn in as captain, get your esword and spacesuit and... go to station to grief. 
Charlie station is at least not antag affiliated, loot is suck-er and not instant to loot

EgorDinamit commented:
> So you just spawn in as captain, get your esword and spacesuit and... go to station to grief.
> Charlie station is at least not antag affiliated, loot is suck-er and not instant to loot

The ship starts unpowered, with 5 tough NPCs which require you to waste your ammo(or to die once) in order to kill them. 

Charlie station can do the same trick easily, it just requires some ghost scouting to find space loot.

I'm trying my best to design this ruin in a way, to prevent people from going to station out of boredom. If it doesn't work, well, then they should check their flavour text once again and get the bwoink.

Fikou commented:
even reflector vests/jackets have 60 laser armor but this has 70

EgorDinamit commented:
> even reflector vests/jackets have 60 laser armor but this has 70

reflector vest can reflect lasers, this one can't.

LemonInTheDark commented:
This man is totally not holding me hostage, I have not been bamboozled into giving a good review.
I like the way this looks, how long do you estimate a ghost role could play here?
Fix the dmi conflicts or the deal's off.
I see you've modified  code/modules/mob/living/simple_animal/hostile/nanotrasen.dm, are these mobs used anywhere else in game? Consider making the behavior you want a subtype, rather then changing the parent.

Good first PR.

ATH1909 commented:
> This man is totally not holding me hostage, I have not been bamboozled into giving a good review.
> I like the way this looks, how long do you estimate a ghost role could play here?
> Fix the dmi conflicts or the deal's off.
> I see you've modified code/modules/mob/living/simple_animal/hostile/nanotrasen.dm, are these mobs used anywhere else in game? Consider making the behavior you want a subtype, rather then changing the parent.
> 
> Good first PR.

Aren't they used in the gene vault away mission? Being able to use the guns of the NPC security team members that you kill is a pretty important part of getting to the "end" of that away mission as a non-cyborg.

ATH1909 commented on code/modules/ruins/spaceruin_code/forgottenship.dm:
```diff
@@ -0,0 +1,115 @@
+	item_state = "cybersun"
+	hardsuit_type = "cybersun"
+	name = "Cybersun hardsuit"
+	desc = "Prototype hardsuit with experimental armor plates, protecting from laser-based weapons very well, while giving limited protection against anything else. Requires the user to activate the inner mechanism in order to unequip it, making it nearly impossible to take it off from somebody else."
```
you should specify that it is still POSSIBLE to remove it from someone via the strip menu, even if it'll take a minute or so to do so (unlike, say, the abductor agent's suit if they haven't unlocked it)

EgorDinamit replied:
Sure thing, but that's just a description after all.

ATH1909 commented on code/modules/ruins/spaceruin_code/forgottenship.dm:
```diff
@@ -0,0 +1,115 @@
+GLOBAL_VAR_INIT(fscpassword, generate_password())
+
+/proc/generate_password()
+	return "[pick(GLOB.phonetic_alphabet)] [rand(1,1000)]"
```
By the way, are you sure that saying "Alpha 1000" won't open the door if the password is "Alpha 100"? I genuinely don

Honestly, that's such an edge case that I wouldn't bother addressing it (and addressing it would probably make it impossible for the door to open for a message if the message is embedded in another sentence (for example: "Door, open. Authorization code: Alpha 100.").

EgorDinamit replied:
I was testing it, saying only the code itself works.
You can also just input it by using the blast-door.

EgorDinamit replied:
I guess it requires to say the specific code, 100 won't work if it's 1000.

ATH1909 commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -588,3 +588,75 @@
+	icon = 'icons/obj/machines/sleeper.dmi'
+	icon_state = "sleeper_s"
+	short_desc = "You are the captain of an old ship, stuck in hostile space."
+	flavour_text = "Your ship docks after a long time somewhere in hostile space, reporting a malfunction. You are stuck here, with Nanotrasen station nearby. Command your crew and turn your ship into the most protected fortress."
```
Is the captain allowed to raid the vault for the cool hardsuits himself?

Is he allowed to add the password on the paper to his notes menu, then burn it so that no one else can read it?

EgorDinamit replied:
Opening the vault is a point of no-return, you can't lock it once you opened it.
The vault itself is made to store secure loot while there is no active captain.

EgorDinamit replied:
> Maybe you should add a GPS signal to this ruin (that'll only appear if the captain spawns, sort of like the comms base GPS signal?) to increase the likelyhood that the syndicate crew members will have someone to fight?

I was thinking of it, but the best thing I could think of is giving the captain with tracking implant on spawn, without teleportation part.

ATH1909 commented:
Maybe you should add a GPS signal to this ruin (that'll only appear if the captain spawns, sort of like the comms base GPS signal?) to increase the likelyhood that the syndicate crew members will have someone to fight?

ATH1909 commented:
Do these guys have access to a full set of tools to conduct repairs with? I'm kind of wondering if there's much to do on this away mission after you clear out the NT NPCs.

cacogen commented:
Seems boring. CharlieStation has a lot to do if you can be bothered.

Are the NT NPCs hostile towards crewmembers who stumble across this?

EgorDinamit commented:
> Seems boring. Charlie/Delta Station has a lot to do if you can be bothered. Also not a fan of those hardsuit sprites.
> 
> Are the NT NPCs hostile towards crewmembers who stumble across this?

NT NPCs have death squad faction so yes, they will attack random people that got on syndicate ship.

EgorDinamit commented:
> This man is totally not holding me hostage, I have not been bamboozled into giving a good review.
> I like the way this looks, how long do you estimate a ghost role could play here?
> Fix the dmi conflicts or the deal's off.
> I see you've modified code/modules/mob/living/simple_animal/hostile/nanotrasen.dm, are these mobs used anywhere else in game? Consider making the behavior you want a subtype, rather then changing the parent.
> 
> Good first PR.

NT NPCs were leaving invisible corpse on death, I guess I fixed it and added some logic to their actions.
They aren't used anywhere in the game, except away missions, I guess.

EgorDinamit committed:
Merge branch 'master' of git://github.com/tgstation/tgstation

EgorDinamit committed:
Changes cybersun hardsuit description

EgorDinamit committed:
Merge branch 'master' into master

EgorDinamit committed:
Merge branch 'master' into master

EgorDinamit committed:
Merge branch 'master' into master

EgorDinamit committed:
Updated icons, so it won't conflict with mafia PR

EgorDinamit committed:
Restoring dme file to what's in main repo.

EgorDinamit committed:
Update tgstation.dme

EgorDinamit committed:
Merge branch 'master' of https://github.com/EgorDinamit/tgstation-1

EgorDinamit committed:
*sigh*

EgorDinamit committed:
Update tgstation.dme

actioninja committed:
fix line endings

EgorDinamit committed:
Merge branch 'master' of https://github.com/EgorDinamit/tgstation-1

EgorDinamit committed:
Merge pull request #5 from tgstation/master

Pull from main tgstation repo.

EgorDinamit committed:
sigh x2

EgorDinamit committed:
Revert "sigh x2"

This reverts commit dbdf53f6e07748a3b47085f5fad9ae586543f127.

EgorDinamit committed:
Update tgstation.dme

EgorDinamit committed:
Revert "Update tgstation.dme"

This reverts commit 2c544d0cb32176290ef2ff346b7709d1efc5389e.

EgorDinamit committed:
Merge branch 'master' into master

EgorDinamit committed:
Update tgstation.dme

EgorDinamit committed:
/

Fikou commented:
> > even reflector vests/jackets have 60 laser armor but this has 70
> 
> Reflector vests can reflect lasers, this one can't.
> Also, worth noting, you **can** buy reflector vests from cargo, while you can only get 2 cybersun hardsuits.

its literally the anti energy armor and youre making a random space suit the best anti laser thing

EgorDinamit commented:
> > > even reflector vests/jackets have 60 laser armor but this has 70
> > 
> > 
> > Reflector vests can reflect lasers, this one can't.
> > Also, worth noting, you **can** buy reflector vests from cargo, while you can only get 2 cybersun hardsuits.
> 
> its literally the anti energy armor and youre making a random space suit the best anti laser thing

Two of them. In random place in space. Which has to be powered and protected at this point. Ruin might not spawn at all.

Maybe I'll think about rebalancing it, but I think it's a reasonable amount as it is.

EgorDinamit committed:
Merge branch 'master' into master

improvedname commented:
Spaceruins:gateaway edition

EgorDinamit commented:
> Spaceruins:gateaway edition

Gateway edition?

improvedname commented:
Aka how all gateaway missions always got powergamed i mean Syndi NPC ai's are easily manipulated especially the ranged ones as they walk forward fire 3 shots then move back. its easy to get it but you can also just weld trough the wall to get it without any trouble

EgorDinamit commented:
> Aka how all gateaway missions always got powergamed i mean Syndi NPC ai's are easily manipulated especially the ranged ones as they walk forward fire 3 shots then move back. its easy to get it but you can also just weld trough the wall to get it without any trouble

Those are NT npcs and they are a lot better in killing people, just trust me.

improvedname commented:
Still the same AI lol

EgorDinamit commented:
> Still the same AI lol

Well, fighting with them in melee is a bad idea.

improvedname commented:
Just throw floortiles 15 brute

EgorDinamit committed:
Update tgstation.dme

EgorDinamit committed:
Merge branch 'master' of https://github.com/EgorDinamit/tgstation-1

EgorDinamit committed:
Update tgstation.dme

EgorDinamit committed:
Fixing merge conflict with suits

EgorDinamit committed:
Merge branch 'master' of git://github.com/tgstation/tgstation

EgorDinamit committed:
Returning cybersun icons after merge conflict

Fikou commented on code/game/objects/items/cards_ids.dm:
```diff
@@ -461,6 +461,26 @@ update_label()
 	access = list(ACCESS_SYNDICATE)
 	uses_overlays = FALSE
 
+/obj/item/card/id/syndicate_command/crew_id
```
why do this when it inherits everything from parent and the captain one inherits everything but the name and assignment

EgorDinamit replied:
For some reason, I thought this would be a better decision.
I'm actually thinking about adding some extra access, so people with syndicate chameleon ID won't be able to get here since lore-wise it's an old ship.

EgorDinamit committed:
Merge branch 'master' into master

EgorDinamit committed:
Merge branch 'master' of git://github.com/tgstation/tgstation

EgorDinamit committed:
Cybersun icons, again.

EgorDinamit committed:
Merge branch 'master' into master

EgorDinamit committed:
Merge pull request #8 from tgstation/master

Fixing merge problems

EgorDinamit committed:
Update forgottenship.dmm

EgorDinamit committed:
Merge branch 'master' of https://github.com/EgorDinamit/tgstation-1

EgorDinamit committed:
Update tgstation.dme

EgorDinamit committed:
Merge branch 'master' of git://github.com/tgstation/tgstation

EgorDinamit committed:
Fixing merge conflicts

actioninja commented:
Fix the conflict and I'll hit this.

EgorDinamit committed:
Merge branch 'master' of git://github.com/tgstation/tgstation

EgorDinamit committed:
Icons fix

EgorDinamit committed:
Merge branch 'master' of git://github.com/tgstation/tgstation

EgorDinamit committed:
Fixing icons, again and again...

actioninja committed:
whitespace fixes

Rohesie commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -640,3 +640,75 @@
+
+/obj/effect/mob_spawn/human/syndicatespace/Destroy()
+	new/obj/structure/fluff/empty_sleeper/syndicate(get_turf(src))
+	..()
```
```suggestion
	return ..()
```

Rohesie commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -640,3 +640,75 @@
+	outfit = /datum/outfit/syndicatespace/syndicrew
+
+/datum/outfit/syndicatespace/syndicrew/post_equip(mob/living/carbon/human/H)
+	H.faction |= ROLE_SYNDICATE
```
Should this be here, in the outfit datum?

EgorDinamit replied:
Dunno, I was quite literally copying lavaland spawner code.

Rohesie commented on code/modules/mob/living/simple_animal/hostile/nanotrasen.dm:
```diff
@@ -24,15 +24,18 @@
 	unsuitable_atmos_damage = 15
-	faction = list("nanotrasenprivate")
+	faction = list(ROLE_DEATHSQUAD)
+	check_friendly_fire = 1
```
```suggestion
	check_friendly_fire = TRUE
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/nanotrasen.dm:
```diff
@@ -24,15 +24,18 @@
-	speak = list("Stop resisting!", "I AM THE LAW!", "Face the wrath of the golden bolt!", "Stop breaking the law, asshole!")
-	search_objects = 1
-
+	del_on_death = 1
```
```suggestion
	del_on_death = TRUE
```

EgorDinamit replied:
By the way, there is a difference from "1" and "true"?

Rohesie commented on code/modules/ruins/spaceruin_code/forgottenship.dm:
```diff
@@ -0,0 +1,115 @@
+GLOBAL_VAR_INIT(fscpassword, generate_password())
+
+/proc/generate_password()
+	return "[pick(GLOB.phonetic_alphabet)] [rand(1,1000)]"
```
```suggestion
	return "[pick(GLOB.phonetic_alphabet)] [rand(1000,9999)]"
```

EgorDinamit replied:
This might sound better, I guess, but I don't remember if, for example, PDA Uplinks had passcode numbers longer than 1000.

Rohesie commented on code/modules/ruins/spaceruin_code/forgottenship.dm:
```diff
@@ -0,0 +1,115 @@
+	icon_living = "nanotrasenrangedassault"
+	icon_dead = null
+	icon_gib = "syndicate_gib"
+	ranged = 1
```
```suggestion
	ranged = TRUE
```

Rohesie commented on code/modules/ruins/spaceruin_code/forgottenship.dm:
```diff
@@ -0,0 +1,115 @@
+	health = 150
+	melee_damage_lower = 13
+	melee_damage_upper = 18
+	ranged = 1
```
```suggestion
	ranged = TRUE
```

actions-user committed:
Automatic changelog compile [ci skip]

EgorDinamit committed:
Update code/modules/mob/living/simple_animal/hostile/nanotrasen.dm

Co-Authored-By: Rohesie <rohesie@gmail.com>

EgorDinamit committed:
Update code/modules/mob/living/simple_animal/hostile/nanotrasen.dm

Co-Authored-By: Rohesie <rohesie@gmail.com>

EgorDinamit committed:
Update code/modules/ruins/spaceruin_code/forgottenship.dm

Co-Authored-By: Rohesie <rohesie@gmail.com>

EgorDinamit committed:
Update code/modules/ruins/spaceruin_code/forgottenship.dm

Co-Authored-By: Rohesie <rohesie@gmail.com>

EgorDinamit committed:
Update code/game/objects/structures/ghost_role_spawners.dm

Co-Authored-By: Rohesie <rohesie@gmail.com>

EgorDinamit committed:
Revert "Automatic changelog compile [ci skip]"

This reverts commit cf43b1858693f924f898f65322eb3df4c5709317.

EgorDinamit committed:
Update code/modules/ruins/spaceruin_code/forgottenship.dm

Co-Authored-By: Rohesie <rohesie@gmail.com>

EgorDinamit committed:
Merge branch 'master' of git://github.com/tgstation/tgstation

EgorDinamit committed:
Update hats.dmi

EgorDinamit committed:
Merge branch 'master' of https://github.com/EgorDinamit/tgstation-1

GravitronBlue commented:
Expected this PR to meet a similar fate to similar ones made by WJohn such as #21977.
That said I don't have much against player syndicate space ruins so long as they generally keep to themselves and away from the station.

My complaints so far with this are focused around the ship design itself and the Nanotrasen elite having the pulse rifle sound but still firing lasers. Critical issues like the lack of parts to build RnD machines is expected to be fixed soon in another PR so I won't bother with those.

May find it worthwhile to check out some of WJohn's work like with the Syndicate Battle Cruiser.
https://i.imgur.com/bLFWIGv.png

adamkad1 commented:
The Cyber Sun suits dont have button for thermal regulation thingy
Also just adding that the system is dumb and shouldnt exist

Rohesie commented:
Being fixed in #50120
